### ISSUE TITLE
Elevation tiles cannot be served from read-only filesystem

### ISSUE THREAD

**Issue by irees** [valhalla/valhalla#1461]
Hi all,

I'd like to keep my elevation tiles on a read-only filesystem since it is quite large and rarely changes, but Valhalla requires elevation tiles to be opened read/write.

Example, with /data mounted read-only:
```
root@43becb83f21c:/build# valhalla_build_config --additional-data-elevation=/data/elevation > valhalla.json
root@43becb83f21c:/build# valhalla_service valhalla.json
2018/08/02 21:42:52.721770 [WARN] Tile extract could not be loaded
terminate called after throwing an instance of 'std::runtime_error'
  what():  /data/elevation/N37/N37W123.hgt(open): Read-only file system
Aborted
```

**Comment by kevinkreiser** [comment]
yeah, thats because we `mmap` with `write` set. for that usecase we could `mmap` with it set to `read-only` but the default is with `write` because we use the same little `mmap` wrapper to do data operations when building tiles. i can look at switching between the two modes based on a parameter for example.

**Comment by irees** [comment]
@kevinkreiser A flag, or having the elevation server/tile builder open in read only, would be a great addition.

**Comment by ManuelLR** [comment]
Any progress on this?
